### PR TITLE
Brings Azure object storage more inline with S3 and GCP.

### PR DESCRIPTION
**Pull request by maestro957**
The injected storage service uses a Singleton to construct the appropriate storage class from the factory. Consequently, the storage parsing was not using the inherited class in Azure so the code would fail looking for azure://. Additionally, the parsing needs to not URL decode the connection string, but the variable to conditionally check that was not available at the appropriate level.

### CODE REVIEW

**Review by pdeffendol — CHANGES_REQUESTED** [review]
Boolean flags in public method signatures can frequently be considered a code smell.  It makes behavior harder to understand, and now that flag behavior has crept into more parts of the library.   I see that it was added in a previous PR, so I don't know the history behind it, but it may warrant a discussion.

**Comment by pdeffendol** [comment]
Do we ever NOT want to URL-decode parts of the connection string?

**Comment by maestro957** [comment]
> Do we ever NOT want to URL-decode parts of the connection string?

Yes. The URL Decode is replacing the Base64 + with spaces and messing that up. I can change the Azure code to do a string replace on that field. Doing that would remove the necessity of having that Boolean parameter.

File: src/Centeva.ObjectStorage.Azure/AzureConnectionFactory.cs [hunk 1]
```diff
@@ -5,21 +5,11 @@ namespace Centeva.ObjectStorage.Azure;
...
-            return null;
-
-        var container = connectionString.GetRequired(Container);
-        var accountName = connectionString.GetRequired(AccountName);
-        var accountKey = connectionString.GetRequired(AccountKey);
-
-        return new AzureObjectStorage(container, connectionString as AzureObjectStorageConnectionString, accountName, accountKey);
+        return connectionString.ProviderName != ProviderName
+            ? null
+            : (IObjectStorage)new AzureObjectStorage(connectionString);
```
Comment (pdeffendol) [inline src/Centeva.ObjectStorage.Azure/AzureConnectionFactory.cs:13]: This cast is not necessary?

File: src/Centeva.ObjectStorage.Azure/AzureObjectStorage.cs [hunk 2]
```diff
@@ -3,33 +3,29 @@
...
 public class AzureObjectStorage : ISignedUrlObjectStorage
 {
     private readonly BlobServiceClient _client;
-    private readonly AzureObjectStorageConnectionString _connectionString;
-    private readonly string _containerName;
+    private readonly ObjectStorageConnectionString _connectionString;
+    private string? _containerName = null;
 
-    public AzureObjectStorage(string containerName, AzureObjectStorageConnectionString? connectionString, string accountName, string accountKey)
+    public AzureObjectStorage(ObjectStorageConnectionString? connectionString)
```
Comment (pdeffendol) [inline src/Centeva.ObjectStorage.Azure/AzureObjectStorage.cs:16]: The other providers (Google, S3) deliberately did not allow the `ObjectStorageConnectionString` to be passed into their constructors.  Parsing the checking for required components was done in the factory classes.  Was there something preventing that from happening here?

**Review by pdeffendol — CHANGES_REQUESTED** [review]

**Comment by pdeffendol** [comment]
> > Do we ever NOT want to URL-decode parts of the connection string?
> 
> Yes. The URL Decode is replacing the Base64 string '+' with spaces and messing that up. I can change the Azure code to do a string replace on that field. Doing that would remove the necessity of having that Boolean parameter.

In that case, change the `Parameters` in `ObjectStorageConnectionString` back to private as it was previously.

File: src/Centeva.ObjectStorage.Azure/AzureObjectStorage.cs [hunk 3]
```diff
@@ -112,4 +96,18 @@ await _client
             .UploadBlobAsync(objectName, dataStream, cancellationToken)
             .ConfigureAwait(false);
     }
+
+    private static string ExtractContainerName(ref string objectName)
+    {
+        if (objectName.Length == 0)
+            return objectName;
+
+        var index = objectName.IndexOf('/');
```
Comment (pdeffendol) [inline src/Centeva.ObjectStorage.Azure/AzureObjectStorage.cs:105]: You could potentially use `StoragePath.Split` here

Comment (pdeffendol) [reply src/Centeva.ObjectStorage.Azure/AzureObjectStorage.cs:105]: or not, it's a private method.  Needs more thought...

**Review by pdeffendol — APPROVED** [review]

**Pull request merged**